src/lib.rs

```diff
@@ -533,15 +550,23 @@ where
             .map_err(Error::clone)
     }
 
-    fn parse_functions(
+    fn parse_functions(&self, sections: &gimli::Dwarf<R>) -> Result<&Functions<R>, Error> {
+        self.funcs
+            .borrow_with(|| Functions::parse(&self.dw_unit, sections))
+            .as_ref()
+            .map_err(Error::clone)
+    }
+
+    fn parse_inlined_functions(
         &self,
         sections: &gimli::Dwarf<R>,
         units: &[ResUnit<R>],
-    ) -> Result<&Functions<R>, Error> {
+    ) -> Result<(), Error> {
         self.funcs
-            .borrow_with(|| Functions::parse(&self.dw_unit, sections, units))
+            .borrow_with(|| Functions::parse(&self.dw_unit, sections))
             .as_ref()
-            .map_err(Error::clone)
+            .map_err(Error::clone)?
```

Comment:
Oh good, this is better than what I did because you store the error, rather than re-parsing errored functions on every call.